impeller/compiler/reflector.cc

```diff
@@ -603,6 +603,29 @@ std::vector<StructMember> Reflector::ReadStructMembers(
 
     FML_CHECK(current_byte_offset == struct_member_offset);
 
+    // A user defined struct.
+    if (member.basetype == spirv_cross::SPIRType::BaseType::Struct) {
+      const size_t size =
+          GetReflectedStructSize(ReadStructMembers(member.self));
+      uint32_t stride = GetArrayStride<0>(struct_type, member, i);
+      if (stride == 0) {
+        stride = size;
+      }
+      uint32_t element_padding = stride - size;
+      result.emplace_back(StructMember{
+          compiler_->get_name(member.self),      // type
+          BaseTypeToString(member.basetype),     // basetype
+          GetMemberNameAtIndex(struct_type, i),  // name
+          struct_member_offset,                  // offset
+          size,                                  // size
+          stride * array_elements.value_or(1),   // byte_length
+          array_elements,                        // array_elements
+          element_padding,                       // element_padding
+      });
+      current_byte_offset += stride * array_elements.value_or(1);
+      continue;
+    }
```

Comment:
I think this adds support for nested structs across all backend and stage types? Might be worth adding a small raster test that uses a nested struct (and make sure the GLES metadata doesn't need to be updated for this).

Reply:
Proooobably?

I'll give it a try.

Reply:
I'm going to punt on this fornow - I'm still trying to track down what to do about the buffer to texture issue and want to keep making progress on compute right now.